sql/scan.go

```diff
@@ -112,31 +106,49 @@ func (n *scanNode) Next() bool {
 			return false
 		}
 
+		var vals valMap
 		if n.primaryKey == nil {
 			// This is the first key for the row, reset our vals map.
-			n.vals = valMap{}
+			vals = valMap{}
```

Comment:
might be cleaner to always create the map but only init it `if n.primaryKey == nil`- it would allow you to avoid the new nil checks in `decodeIndexKey`